### ISSUE TITLE
com.sun.deploy.uitoolkit.impl.fx.HostServicesFactory class missing

### ISSUE THREAD

**Issue by Sabir001** [ojdkbuild/ojdkbuild#68]
When I run a javafx program by using opnejdk 8, I get this exception,

`java.lang.ClassNotFoundException: com.sun.deploy.uitoolkit.impl.fx.HostServicesFactory`

**Comment by ojdkbuild** [comment]
Hi, off the top of my head: the deployment part of the OpenJFX8 is tied with Oracle Deployment stack - plugin and webstart, not open-source. These class you need is probably comes from there. Another deployment feature that will not work with OpenJFX8 is [fx:deploy ant task](https://github.com/ojdkbuild/upstream_openjfx-8u/commit/0ec04d783e39b5a6523bab6b2be56090f52f4051).

Generally I need some way to reproduce this to investigate it.

**Comment by Sabir001** [comment]
```
<plugins>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-dependency-plugin</artifactId>
                <version>2.6</version>
                <executions>
                    <execution>
                        <id>unpack-dependencies</id>
                        <phase>package</phase>
                        <goals>
                            <goal>unpack-dependencies</goal>
                        </goals>
                        <configuration>
                            <excludeScope>system</excludeScope>
                            <excludeGroupIds>junit,org.mockito,org.hamcrest</excludeGroupIds>
                            <outputDirectory>${project.build.directory}/classes</outputDirectory>
                        </configuration>
                    </execution>
                </executions>
            </plugin>
            <plugin>
                <groupId>org.codehaus.mojo</groupId>
                <artifactId>exec-maven-plugin</artifactId>
                <version>1.2.1</version>
                <executions>
                    <execution>
                        <id>unpack-dependencies</id>
                        
                        <phase>package</phase>
                        <goals>
                            <goal>exec</goal>
                        </goals>
                        <configuration>
                            <executable>${java.home}/../bin/javafxpackager</executable>
                            <arguments>
                                <argument>-createjar</argument>
                                <argument>-nocss2bin</argument>
                                <argument>-appclass</argument>
                                <argument>${mainClass}</argument>
                                <argument>-srcdir</argument>
                                <argument>${project.build.directory}/classes</argument>
                                <argument>-outdir</argument>
                                <argument>${project.build.directory}</argument>
                                <argument>-outfile</argument>
                                <argument>${project.build.finalName}.jar</argument>
                            </arguments>
                        </configuration>
                    </execution>
                    <execution>
                        <id>default-cli</id>
                        <goals>
                            <goal>exec</goal>                            
                        </goals>
                        <configuration>
                            <executable>${java.home}/bin/java</executable>
                            <commandlineArgs>${runfx.args}</commandlineArgs>
                        </configuration>
                    </execution>
                </executions>  
            </plugin>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-compiler-plugin</artifactId>
                <version>3.1</version>
                <configuration>
                    <source>1.8</source>
                    <target>1.8</target>
                    <compilerArguments>
                        <bootclasspath>${sun.boot.class.path}${path.separator}${java.home}/lib/jfxrt.jar</bootclasspath>
                    </compilerArguments>
                </configuration>
            </plugin>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-surefire-plugin</artifactId>
                <version>2.16</version>
                <configuration>
                    <additionalClasspathElements>
                        <additionalClasspathElement>${java.home}/lib/jfxrt.jar</additionalClasspathElement>
                    </additionalClasspathElements>
                </configuration>
            </plugin>
        </plugins>
```

This are the plugins I am using. Basically when I create a maven javafx project using Apache Netbeans 10, these are already there in` pom.xml` file. Hope this will help you to reproduce.

**Comment by Sabir001** [comment]
This problem occurs when I try to set my `HostServices ` with `getHostServices()` method. 
I need this as I want to open some pages in browser with my app.

**Comment by Sabir001** [comment]
Take a look at [this issue](https://bugs.openjdk.java.net/browse/JDK-8160464)

**Comment by ojdkbuild** [comment]
Thanks for the link! I will look into it.

**Comment by ojdkbuild** [comment]
The fix from [JDK-8160464](https://bugs.openjdk.java.net/browse/JDK-8160464) is included with [8u212 release](https://github.com/ojdkbuild/ojdkbuild/releases/tag/1.8.0.212-1) (see openjfx-8.0.202-1.b08 binaries). I am a bit confused, why is it marked 8-N/A as it seems to be applicable to 8. Closing the issue, please re-open if it won't help.
